### PR TITLE
[CARBONDATA-775]Updated Date DataType in DataTypes supported by CarbonData

### PR DESCRIPTION
**Pull request by PallaviSingh1992**
Added the support for Date Data Type and re-structured the file.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1158/

**Comment by PallaviSingh1992** [comment]
@chenliang613 please review

File: docs/supported-data-types-in-carbondata.md [hunk 1]
```diff
@@ -22,17 +22,18 @@
...
+    * DECIMAL
 
   * Date/Time Types
-  * TIMESTAMP
+    * TIMESTAMP
+    * DATE
 
   * String Types
-  * STRING
+    * STRING
```
Comment (jackylk) [inline docs/supported-data-types-in-carbondata.md]: I think CHAR can also be added

Comment (PallaviSingh1992) [inline docs/supported-data-types-in-carbondata.md]: @jackylk Thanks, resolved the review comment

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1187/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1203/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**